lib/src/index_impl.dart

```diff
@@ -53,17 +53,17 @@ class MeiliSearchIndexImpl implements MeiliSearchIndex {
 
   factory MeiliSearchIndexImpl.fromMap(
     MeiliSearchClient client,
-    Map<String, dynamic>? map,
+    Map<String, Object?>? map,
   ) =>
       MeiliSearchIndexImpl(
         client,
         map?['uid'] as String,
         primaryKey: map?['primaryKey'] as String?,
-        createdAt: map?['createdAt'] != null
-            ? DateTime.tryParse(map?['createdAt'] as String)
+        createdAt: map?['createdAt'] is String
+            ? DateTime.tryParse(map!['createdAt'] as String)
```

Comment:
unfortunately yes, this is a known limitation in dart, it can't promote nullable values that aren't in local context.
an alternative would be:
```dart
final createdAt = map?['createdAt'];
//...
createdAt: createdAt is String ? DateTime.tryParse(createdAt) : null
```

Reply:
That's unfortunate. In any case, I think you can keep the current version.